### DISCUSSION TITLE
how to limit resource for single image request，not limit global

### DISCUSSION THREAD

**Discussion by mh47838704** [ImageMagick/ImageMagick#2373] [Help]
first，i want limit the resource of parts of request not for all， because gif type resize or other operation consume large part of mem and cpu(with muti thread)。

second，i want to know how to optimization gif deal，consume less mem and cpu resource，just like **gifscile**，it fast and consume little mem and cpu resource，event the result is not good as imagemagick，but i can accept。i believe imagemagick can reach the same effect。

in imagemagick document，resource limit is global，set limit will affect all image deal

**Comment by snibgo** [comment]
What version of IM, on what platform?

At the command line, you can limit resources, eg "-limit memory 1MB". You can set a small limit before a particular operation, then set a larger limit after the operation.

You can't use this to set limits larger than those in policy.xml.

**Reply by mh47838704** [reply]
my version:ImageMagick 7.0.8-60 Q16 x86_64，not command line

my program use gographics integrated imagemagic lib，my program run as webserver， is a single process ，serve each request with a goroutine， if i use interface SetResourceLimit to set resource，it affect all request， all goroutine share the same config。（but this is not what i want）

what i want is，set limit for each request，so i can controll more flexible

but it looks like the config is static ，so all the thread will share

**Comment by snibgo** [comment]
Sorry, I know nothing about gographics.

**Reply by mh47838704** [reply]
tks,too

i try the cmd way, run two "convert .....limit....", memory  limit is different , the effect is expected 
i try the integration way, set memory limit one time , all request share the same cache

so the cache is process level,  different  process different limit, but to thread, all share the same cache,
